support/sql/performance/resolve-blocking-problems-caused-lock-escalation.md

```diff
@@ -33,17 +33,17 @@ The simplest and safest way to prevent lock escalation is to keep transactions s
 - Break up large batch operations into several smaller operations. For example, suppose you ran the following query to remove several hundred thousand old records from an audit table, and then you found that it caused a lock escalation that blocked other users:
 
     ```sql
-    DELETE FROM LogMessages WHERE LogDate < '2/1/2002'
+    DELETE FROM LogMessages WHERE LogDate < '20020102';
     ```
 
   By removing these records a few hundred at a time, you can dramatically reduce the number of locks that accumulate per transaction and prevent lock escalation. For example:
 
     ```sql
-    SET ROWCOUNT 500
-    delete_more:
-    DELETE FROM LogMessages WHERE LogDate < '2/1/2002'
-    IF @@ROWCOUNT > 0 GOTO delete_more
-    SET ROWCOUNT 0
+    WHILE (1 = 1)
+    BEGIN
+        DELETE TOP(500) FROM LogMessages WHERE LogDate < '20020102';
+        IF @@ROWCOUNT < 500 BREAK;
```

Comment:
```suggestion
        SET @rows = @@ROWCOUNT;
```

Reply:
True that SET ROWCOUNT is deprecated for DELETE statements, but using BREAK in this type of situation is not recommended.